packages/experience-builder-sdk/src/hooks/useFetchExperience.ts

```diff
@@ -144,21 +159,34 @@ export const useFetchExperience = ({ mode, client }: useClientsideExperienceFetc
           setExperience(experience);
 
           return experience;
-        } catch (e) {
-          handleError(errorMessagesWhileFetching.experienceReferences, e);
-          throw e;
+        } catch (error) {
+          handleError(errorMessagesWhileFetching.experienceReferences, error);
+          setError(error as Error);
         }
-      } catch (e) {
-        handleError(errorMessagesWhileFetching.experience, e);
-        throw e;
+      } catch (error) {
+        handleError(errorMessagesWhileFetching.experience, error);
+        setError(error as Error);
       } finally {
         setIsFetching(false);
       }
     },
     [client, mode]
   );
 
+  useEffect(() => {
+    if (slug) {
+      fetchBySlug({ slug, localeCode, experienceTypeId });
+      return;
+    }
+    if (id) {
```

Comment:
Nit: Wonder if the order matters, what if both slug and ID are provided for some reason, wouldn't we prefer fetching by ID?